Errors building on OS X 10.10.5

<!--
Thank you for reporting an issue.
Please fill in as much of the template below as you're able.

Version: output of `node -v`
Platform: output of `uname -a` (UNIX), or version and 32 or 64-bit (Windows)
Subsystem: if known, please specify affected core module name

If possible, please provide code that demonstrates the problem, keeping it as
simple and free of external dependencies as you are able.
-->
- **Version**: `xplat`
- **Platform**: OS X 10.10.5, XCode 7.2
- **Subsystem**: build

<!-- Enter your issue details below this comment. -->

```
./configure --engine=chakracore && make -j8 test
```

```
[ 21%] Building CXX object lib/Common/Memory/CMakeFiles/Chakra.Common.Memory.dir/LeakReport.cpp.o
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/System/Library/Frameworks/Security.framework/Headers/CSCommon.h:200:32: error: enumerator value evaluates to -2147483648, which cannot be narrowed to type 'uint32_t' (aka 'unsigned int') [-Wc++11-narrowing]
    kSecCSConsiderExpiration = 1 << 31,
                               ^
1 error generated.
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/System/Library/Frameworks/Security.framework/Headers/CSCommon.h:200:32: error: enumerator value evaluates to -2147483648, which cannot be narrowed to type 'uint32_t' (aka 'unsigned int') [-Wc++11-narrowing]
    kSecCSConsiderExpiration = 1 << 31,
                               ^
1 error generated.
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/System/Library/Frameworks/Security.framework/Headers/CSCommon.h:200:32: error: enumerator value evaluates to -2147483648, which cannot be narrowed to type 'uint32_t' (aka 'unsigned int') [-Wc++11-narrowing]
    kSecCSConsiderExpiration = 1 << 31,
                               ^
1 error generated.
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/System/Library/Frameworks/Security.framework/Headers/CSCommon.h:200:32: error: enumerator value evaluates to -2147483648, which cannot be narrowed to type 'uint32_t' (aka 'unsigned int') [-Wc++11-narrowing]
    kSecCSConsiderExpiration = 1 << 31,
                               ^
1 error generated.
make[4]: *** [pal/src/CMakeFiles/Chakra.Pal.dir/cruntime/filecrt.cpp.o] Error 1
make[4]: *** Waiting for unfinished jobs....
make[4]: *** [pal/src/CMakeFiles/Chakra.Pal.dir/cruntime/file.cpp.o] Error 1
make[4]: *** [pal/src/CMakeFiles/Chakra.Pal.dir/cruntime/finite.cpp.o] Error 1
make[4]: *** [pal/src/CMakeFiles/Chakra.Pal.dir/cruntime/lstr.cpp.o] Error 1
[ 21%] Building CXX object lib/Runtime/PlatformAgnostic/Platform/CMakeFiles/Chakra.Runtime.PlatformAgnostic.dir/Linux/UnicodeText.ICU.cpp.o
  cc '-D_REENTRANT' '-DPURIFY' '-DOPENSSL_NO_COMP' '-DOPENSSL_NO_SSL3' '-DOPENSSL_NO_HEARTBEATS' '-DNODE_ENGINE="chakracore"' '-DNODE_ENGINE_CHAKRACORE' '-DENGINESDIR="/dev/null"' '-DTERMIOS' '-DOPENSSLDIR="/System/Library/OpenSSL/"' '-D_DARWIN_USE_64_BIT_INODE=1' '-DL_ENDIAN' '-DAES_ASM' '-DCPUID_ASM' '-DOPENSSL_BN_ASM_MONT' '-DOPENSSL_CPUID_OBJ' '-DSHA1_ASM' '-DSHA256_ASM' '-DSHA512_ASM' '-DGHASH_ASM' '-DVPAES_ASM' '-DBN_ASM' '-DBF_ASM' '-DBNCO_ASM' '-DDES_ASM' '-DLIB_BN_ASM' '-DMD5_ASM' '-DOPENSSL_BN_ASM' '-DRIP_ASM' '-DRMD160_ASM' '-DWHIRLPOOL_ASM' '-DWP_ASM' '-DOPENSSL_BN_ASM_MONT5' '-DOPENSSL_BN_ASM_GF2m' '-DOPENSSL_IA32_SSE2' '-DBSAES_ASM' '-DDSO_DLFCN' '-DHAVE_DLFCN_H' -I../deps/openssl -I../deps/openssl/openssl -I../deps/openssl/openssl/crypto -I../deps/openssl/openssl/crypto/asn1 -I../deps/openssl/openssl/crypto/evp -I../deps/openssl/openssl/crypto/md2 -I../deps/openssl/openssl/crypto/modes -I../deps/openssl/openssl/crypto/store -I../deps/openssl/openssl/include  -Os -gdwarf-2 -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -fno-strict-aliasing -MMD -MF /Users/Jeremiah/Documents/node-chakracore/out/Release/.deps//Users/Jeremiah/Documents/node-chakracore/out/Release/obj.target/openssl/deps/openssl/openssl/crypto/seed/seed_ofb.o.d.raw   -c -o /Users/Jeremiah/Documents/node-chakracore/out/Release/obj.target/openssl/deps/openssl/openssl/crypto/seed/seed_ofb.o ../deps/openssl/openssl/crypto/seed/seed_ofb.c
[ 22%] Building CXX object lib/Common/Memory/CMakeFiles/Chakra.Common.Memory.dir/MarkContext.cpp.o
[ 22%] Building CXX object lib/Runtime/PlatformAgnostic/Platform/CMakeFiles/Chakra.Runtime.PlatformAgnostic.dir/Unix/DateTime.cpp.o
[ 22%] Building CXX object lib/Common/Memory/CMakeFiles/Chakra.Common.Memory.dir/MemoryLogger.cpp.o
[ 23%] Building CXX object lib/Common/Memory/CMakeFiles/Chakra.Common.Memory.dir/MemoryTracking.cpp.o
[ 23%] Building CXX object lib/Runtime/Base/CMakeFiles/Chakra.Runtime.Base.dir/CallInfo.cpp.o
[ 23%] Building CXX object lib/Common/Memory/CMakeFiles/Chakra.Common.Memory.dir/PageAllocator.cpp.o
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/System/Library/Frameworks/Security.framework/Headers/CSCommon.h:200:32: error: enumerator value evaluates to -2147483648, which cannot be narrowed to type 'uint32_t' (aka 'unsigned int') [-Wc++11-narrowing]
    kSecCSConsiderExpiration = 1 << 31,
                               ^
1 error generated.
```

Hmmm, squashing these with `-Wno-c++11-narrowing` seems to work but then I get `-Wkeyword-compat` warnings.

Squashing those leads to LOTS of errors like:

```
In file included from /Users/Jeremiah/Documents/node-chakracore/deps/chakrashim/core/pal/src/include/pal/dbgmsg.h:160:
/Users/Jeremiah/Documents/node-chakracore/deps/chakrashim/core/pal/src/include/pal/thread.hpp:285:16: error: unelaborated friend declaration is a C++11 extension; specify 'class' to befriend 'CatchHardwareExceptionHolder' [-Werror,-Wc++11-extensions]
        friend CatchHardwareExceptionHolder;
               ^
               class
/Users/Jeremiah/Documents/node-chakracore/deps/chakrashim/core/pal/src/include/pal/thread.hpp:751:24: error: use of undeclared identifier 'nullptr'
        if (pThread == nullptr)
                       ^
7 errors generated.
make[4]: *** [pal/src/CMakeFiles/Chakra.Pal.dir/cruntime/finite.cpp.o] Error 1
In file included from /Users/Jeremiah/Documents/node-chakracore/deps/chakrashim/core/pal/src/cruntime/lstr.cpp:22:
In file included from /Users/Jeremiah/Documents/node-chakracore/deps/chakrashim/core/pal/src/include/pal/palinternal.h:320:
In file included from /Users/Jeremiah/Documents/node-chakracore/deps/chakrashim/core/pal/inc/pal.h:72:
/Users/Jeremiah/Documents/node-chakracore/deps/chakrashim/core/pal/inc/pal_mstypes.h:605:9: error: unknown type name 'char16_t'
typedef char16_t __wchar_16_cpp__;
        ^
/Users/Jeremiah/Documents/node-chakracore/deps/chakrashim/core/pal/inc/pal_mstypes.h:612:9: error: unknown type name 'char16_t'
typedef char16_t WCHAR;
        ^
```

@obastemur still getting e.g.

```
[ 12%] Built target Chakra.Common.Common
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/usr/include/c++/4.2.1/bits/cpp_type_traits.h:134:12: error: keyword '__is_void' will be made available as an identifier for the remainder of the translation unit [-Werror,-Wkeyword-compat]
    struct __is_void
           ^
```

Maybe something is wrong with my XCode install? What XCode versions / frameworks are you using?

@obastemur hmmm, it might have progressed further but I'm still getting:

```
[ 31%] Building CXX object lib/Common/Memory/CMakeFiles/Chakra.Common.Memory.dir/HeapBlockMap.cpp.o
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/usr/include/c++/4.2.1/bits/cpp_type_traits.h:134:12: error: keyword '__is_void' will be made available as an identifier for the remainder of the translation unit [-Werror,-Wkeyword-compat]
    struct __is_void
           ^
```

a couple times and 

```
[ 38%] Building CXX object lib/Runtime/Debug/CMakeFiles/Chakra.Runtime.Debug.dir/ProbeContainer.cpp.o
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/System/Library/Frameworks/Security.framework/Headers/CSCommon.h:200:32: error: enumerator value evaluates to -2147483648, which cannot be narrowed to type 'uint32_t' (aka 'unsigned int') [-Wc++11-narrowing]
    kSecCSConsiderExpiration = 1 << 31,
                               ^
```

:/

--- a/deps/chakrashim/chakracore.gyp
+++ b/deps/chakrashim/chakracore.gyp
@@ -11,6 +11,7 @@
     'linker_end_group%': '',
     'chakra_libs_absolute%': '',
     'icu_lib_path%': '',
+    'chakra_build_flags': '',
 
     # xplat (non-win32) only
     'chakra_config': 'Release',     # Debug, Release
@@ -28,12 +29,15 @@
         'icu_include_path': '/usr/local/opt/icu4c/include'
       }],
 
+      # xplat-todo: we need to build CC release/debug without breaking the previous cache
+      # node compiles both release and debug on `debug` mode.
+      # ChakraCore ends-up overwriting the previous build cache
       # xplat (non-win32) only
-      ['chakra_config=="Debug"', {
-        'chakra_build_flags': [ '-d' ],
-      }, {
-        'chakra_build_flags': [],
-      }],
+      # ['chakra_config=="Debug"', {
+      #   'chakra_build_flags': [ '-d' ],
+      # }, {
+      #   'chakra_build_flags': [],
+      # }],
     ],
   },
 
@@ -129,7 +133,6 @@
                 '<(chakra_dir)/build.sh',
                 '--without=Simdjs',
                 '--static',
-                '-j',
                 '<@(chakra_build_flags)',
                 '<@(icu_args)'
               ],

